build-image/boringcrypto/Dockerfile

```diff
@@ -0,0 +1,107 @@
+# syntax=docker/dockerfile:1.4
+
+# NOTE: The main difference in this file vs the parent Dockerfile is using mcr.microsoft.com/oss/go/microsoft/golang:1.22-bullseye which is a microsoft
```

Comment:
I'm worried about this getting desynced over time with the other Linux Dockerfile. Can we find a way to templatize the existing Linux Dockerfile instead of creating a new one? IIUC it's possible to have `ARG` statements which are controllable via the `docker build` command.

Reply:
That worked like a charm, thanks!